synapse/state/__init__.py

```diff
@@ -83,7 +83,7 @@ def _gen_state_id() -> str:
 
 
 class _StateCacheEntry:
-    __slots__ = ["state", "state_group", "state_id", "prev_group", "delta_ids"]
+    __slots__ = ["state", "state_group", "prev_group", "delta_ids"]
```

Comment:
I was slightly surprised to see us use `frozendict` directly below, since there's a config option `use_frozen_dicts`, [default False](https://github.com/matrix-org/synapse/blob/0ef1307619799bec2bbb96ce6516f307b0f8f217/synapse/config/server.py#L262), which I couldn't see documented [here](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html).

Could/should we remove this option? (Not now ofc)

Reply:
`use_frozen_dicts` was more for using frozendict for events rather than here.